web-client/src/components/SignUpForm.jsx

```diff
@@ -0,0 +1,354 @@
+      .max(26, "First name should be 26 characters or less")
+      .required("Required field"),
+    lastName: yup
+      .string()
+      .matches(/^[A-Za-z]*$/, "Last name should only contain letters")
+      .max(26, "Last name should be 26 characters or less")
+      .required("Required field"),
+    email: yup
+      .string()
+      .email("Email should have correct format")
+      .required("Required field"),
+    password: yup
+      .string()
+      .required("Required field")
+      .min(12, "Password must be at least 12 characters long")
+      .matches(
+        /^.*(?=.{12,255})((?=.*[!@#$%^&*()\-_=+{};:,<.>]){1})(?=.*\d)((?=.*[a-z]){1})((?=.*[A-Z]){1}).*$/,
+        "At least: 12 characters, 1 uppercase, 1 lowercase, 1 number and 1 special character"
+      ),
+    university: yup.string().max(255, "University should be 255 character long or less"),
+    academicTitle: yup
+      .string()
+      .max(255, "Academic title should be 255 character long or less"),
+    acceptance: yup.boolean().oneOf([true], "Required field").required("Required field"),
+  });
+
+  const { register, handleSubmit, errors } = useForm({
+    defaultValues: {
+      email: "",
+    },
+    mode: "onBlur",
+    resolver: yupResolver(schema),
+  });
+
+  const handleChange = (prop) => (event) => {
+    setValues({ ...values, [prop]: event.target.value });
+  };
+
+  const handleClickShowPassword = () => {
+    setValues({ ...values, showPassword: !values.showPassword });
+  };
+
+  const handleMouseDownPassword = (event) => {
+    event.preventDefault();
+  };
+
+  const handleChangeSelect = (event) => {
+    setSpecialization(event.target.value);
+  };
+  const style = styles();
+
+  return (
+    <Container component="main">
+      <div className={style.main}>
+        <div classname={style.columns}>
+          <form
+            className={style.form}
+            noValidate
+            onSubmit={handleSubmit((data) => alert(JSON.stringify(data)))}
+          >
+            <TextField
+              className={style.textField}
+              inputRef={register}
+              required
+              id="first-name-signup"
+              name="firstName"
+              label="First name"
+              autoComplete="first-name"
+              InputLabelProps={{
+                shrink: true,
+              }}
+              variant="outlined"
+              error={!!errors.firstName}
+              helperText={errors?.firstName?.message}
+            />
+            <TextField
+              className={style.textField}
+              inputRef={register}
+              required
+              id="last-name-signup"
+              name="lastName"
+              label="Last name"
+              autoComplete="family-name"
+              InputLabelProps={{
+                shrink: true,
+              }}
+              variant="outlined"
+              error={!!errors.lastName}
+              helperText={errors?.lastName?.message}
+            />
```

Comment:
It's hard to read :/ Can you add some comment about what exactly we are looking at? e.g. 'FirstName input'

Reply:
ok, done!